modules/run-consul/run-consul

```diff
@@ -316,14 +316,14 @@ EOF
 }
 
 function generate_supervisor_config {
-  local readonly supervisor_config_path="$1"
-  local readonly consul_config_dir="$2"
-  local readonly consul_data_dir="$3"
-  local readonly consul_log_dir="$4"
-  local readonly consul_bin_dir="$5"
-  local readonly consul_user="$6"
+  local -r supervisor_config_path="$1"
+  local -r consul_config_dir="$2"
+  local -r consul_data_dir="$3"
+  local -r consul_log_dir="$4"
+  local -r consul_bin_dir="$5"
+  local -r consul_user="$6"
   shift 6
-  local readonly environment=("$@")
+  local -r environment=("$@")
```

Comment:
There's one gotcha here: all arrays should use `-ar`. Otherwise, they don't seem to be properly locally scoped in some versions of bash. We had a long discussion in #troubleshooting about this and it took a while to figure that out!